signer/contentsignature/contentsignature_test.go

```diff
@@ -218,3 +219,21 @@ func TestNoShortData(t *testing.T) {
 		t.Fatalf("expected to fail with input data too short but failed with: %v", err)
 	}
 }
+
+// Reproduce nil error
+type ErrorReader struct{}
+
+func (e *ErrorReader) Read(p []byte) (n int, err error) {
+	return 0, errors.New("read error")
+}
+
+func TestRandReadFailureOnSignHash(t *testing.T) {
+	input := []byte("foobarbaz1234abcd")
+	testcase := PASSINGTESTCASES[0]
+	s, _ := New(testcase.cfg)
+	s.rand = &ErrorReader{}
+	_, err_nil := s.SignData(input, nil)
+	if err_nil == nil {
+		t.Fatal("Should have failed to sign data with error in the SignHash.")
+	}
```

Comment:
just `err`, please. There are no underscores in Go code, but also giving errors different names are things we Go hackers do only to prevent shadowing.

Reply:
Good to know. Thanks for sharing this bit on Go best practices.

Resolved [7b19fee](https://github.com/mozilla-services/autograph/pull/1033/commits/7b19fee2a2e3daac50d64af473f12aa4ea1f72c8)